keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlArtifactRepository.kt

```diff
@@ -181,11 +185,62 @@ class SqlArtifactRepository(
         .set(ARTIFACT_VERSIONS.TYPE, type)
         .set(ARTIFACT_VERSIONS.VERSION, version)
         .set(ARTIFACT_VERSIONS.RELEASE_STATUS, status?.toString())
+        .set(ARTIFACT_VERSIONS.BUILD_METADATA, objectMapper.writeValueAsString(artifactMetadata?.buildMetadata))
+        .set(ARTIFACT_VERSIONS.GIT_METADATA, objectMapper.writeValueAsString(artifactMetadata?.gitMetadata))
         .onDuplicateKeyIgnore()
         .execute()
     } == 1
   }
 
+  override fun updateArtifactMetadata(name: String, type: ArtifactType, version: String, status: ArtifactStatus?, artifactMetadata: ArtifactMetadata?) {
```

Comment:
Does it make sense to even call this function if `artifactMetadata` is `null`? It feels to me like that parameter should be non-nullable.

Reply:
It does, because there can be a variety of reasons for the call not to work or some missing params, something is wrong with the artifact structure... I don't want it to fail because it's not nullable

Reply:
Sorry, I don't follow. The function is called `updateArtifactMetadata`, so supposedly you'd only call this when you have some metadata to set, and `artifactMetadata` is not `null`, right? Do you ever expect to call this function to set the metadata to `null`? That's what I was trying to say...

Reply:
ohhh i see what you are saying. will change it.

Reply:
done!